tests/run/builtin_exceptions.py

```diff
@@ -0,0 +1,739 @@
+# mode: run
+# tag: exception
+
+
+NEWER_EXCEPTIONS = {
+    'PythonFinalizationError': (3, 10),
+    'ExceptionGroup': (3, 11),
+    'BaseExceptionGroup': (3, 11),
+    'PythonFinalizationError': (3, 13),
+}
+
+CALL_ARGS = {
+    'UnicodeError'
+}
+
+def _type_name(obj):
+    return type(obj).__name__
+
+
+def gen_tests():
+    import builtins
+    from collections import defaultdict
+
+    with open(__file__, encoding='utf8') as test_file:
+        test_code = test_file.readlines()
+
+    # Note: the following strings must not contain the marker lines!
+    original_end_code = test_code[test_code.index("##### " + "END GENERATED TESTS\n"):]
+    del test_code[test_code.index("##### " + "BEGIN GENERATED TESTS\n") + 1:]
+
+    subclasses = defaultdict(list)
+    for exc_name, exc_type in vars(builtins).items():
+        if not isinstance(exc_type, type) or not issubclass(exc_type, BaseException):
+            continue
+        if exc_name in NEWER_EXCEPTIONS:
+            continue
+        try:
+            exc_type()
+        except TypeError:
+            # Needs arguments - ignore.
+            continue
+        for base_type in exc_type.__mro__:
+            subclasses[base_type.__name__].append(exc_name)
+            if base_type.__name__ == 'BaseException':
+                break
+
+    for exc_name, subclass_names in sorted(subclasses.items()):
+        test_code.append("\n")
+        test_code.append(f"def accept_{exc_name}(exc: {exc_name}):\n")
+
+        func_code = []
+        func_code.append('"""')
+        for subclass_name in subclass_names:
+            func_code.append(f">>> accept_{exc_name}({subclass_name}())")
+        func_code.append('"""')
+        func_code.append(f"assert isinstance(exc, BaseException)")
+        func_code.append(f"assert {exc_name} in type(exc).__mro__")
+        func_code.append(f"assert isinstance(exc, {exc_name})")
+
+        # indent and insert
+        test_code.extend(f"    {line}\n" for line in func_code)
+
+    test_code.append("\n")
+    test_code += original_end_code
+    test_code_str = ''.join(test_code)
+
+    with open(__file__, encoding='utf8', mode='w') as test_file:
+        test_file.write(test_code_str)
+
+##### BEGIN GENERATED TESTS
```

Comment:
The one thing my PR had that yours doesn't is assignment to annotated function-locals.

That generates slightly different type-check code to the argument type check so it might be worth testing here.

Reply:
I extended the tests. It actually does not apply any type checks except for `Exception` and `BaseException` which are specifically declared in `Builtin.py` as builtin extension types.